packages/fiber/src/core/events.ts

```diff
@@ -162,20 +166,16 @@ export function createEvents(store: UseStore<RootState>) {
       const unprojectedPoint = temp.set(mouse.x, mouse.y, 0).unproject(camera)
       const delta = event.type === 'click' ? calculateDistance(event) : 0
       const releasePointerCapture = (id: number) => (event.target as Element).releasePointerCapture(id)
-      const localState = { stopped: false, captured: false }
+
+      const localState = { stopped: false }
 
       for (const hit of intersections) {
+        const hasPointerCapture = (id: number) => internal.capturedMap.get(id)?.eventObject === hit.eventObject
+
         const setPointerCapture = (id: number) => {
-          // If the hit is going to be captured flag that we're in captured state
-          if (!localState.captured) {
-            localState.captured = true
-            // The captured hit array is reset to collect hits
-            internal.captured = []
-          }
-          // Push hits to the array
-          if (internal.captured)
-            internal.captured.push(hit)
-            // Call the original event now
+          // Maybe we should set a warning if the eventId was already captured
+          internal.capturedMap.set(id, hit)
```

Comment:
Not sure if it'll create unwanted side-effects but should we check if the Mapped value exists and if not, set it?

Reply:
Good point, we could. I'll try to see what happens in the DOM when an element tries to capture an event id which is already captured and mimic that behavior.

Reply:
So I checked in the DOM and what happens is that a target can steal a pointerId capture, so there's no need for a check actually, the behavior should be correct here. I'll remove the comment.

Reply:
Might be good to replace the comment with the above so we know why its like the way it is? 🤓

Reply:
That's what I did indeed ;)